client/interfaces.go

```diff
@@ -29,6 +30,7 @@ type Client interface {
 
 // Writer is a client taht sends a batch of samples to remote.
 type Writer interface {
+	WriteTo(samples model.Samples, r *http.Request, w io.Writer) error
```

Comment:
Wonder if there should be only `WriteTo`, or if Write should have a default implementation that calls `WriteTo` by default.

Reply:
We can't have only WriteTo since the destination for regular Write is known only by the writer.
We can't really have a default implementation of Write using WriteTo for the same reason.